SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -124,6 +124,26 @@ void USpatialReceiver::OnAddEntity(const Worker_AddEntityOp& Op)
 	UE_LOG(LogSpatialReceiver, Verbose, TEXT("AddEntity: %lld"), Op.entity_id);
 }
 
+void USpatialReceiver::RemoveRedundantRemoveComponentOps(const Worker_AddComponentOp& Op)
+{
+	int32 RemovedOps = 0;
+	for (int32 i = QueuedRemoveComponentOps.Num() - 1; i >= 0; --i)
+	{
+		const Worker_RemoveComponentOp& RemoveComponentOp = QueuedRemoveComponentOps[i];
+		if (RemoveComponentOp.entity_id == Op.entity_id &&
+			RemoveComponentOp.component_id == Op.data.component_id)
+		{
+			QueuedRemoveComponentOps.RemoveAt(i);
```

Comment:
While this is being changed anyway, it's worth noting that removing the element while iterating over it means you will miss elements. The three ways to get around this are do the index gymnastics for removing RemovedOps from i, make a temporary array of indices and remove them after iterating, or swap the elements to be removed to the back and then bulk remove the end of the array at after iterating. The latter is what most RemoveAll like functions do.

Reply:
Ah wait you are iterating backwards, never mind then. Although I'd still go with the 3rd instead of removing them one by one.